Mathlib/AlgebraicGeometry/Morphisms/Constructors.lean

```diff
@@ -291,6 +291,30 @@ lemma stalkwiseIsLocalAtTarget_of_respectsIso (hP : RingHom.RespectsIso P) :
     exact ((RingHom.toMorphismProperty P).arrow_mk_iso_iff <|
       morphismRestrictStalkMap f (U i) ⟨x, hi⟩).mp <| hf i ⟨x, hi⟩
 
+/-- If `P` respects isos, then `stalkwise P` is local at the source. -/
+lemma stalkwise_isLocalAtSource_of_respectsIso (hP : RingHom.RespectsIso P) :
+    IsLocalAtSource (stalkwise P) := by
+  letI := stalkwise_respectsIso hP
+  apply IsLocalAtSource.mk'
+  · intro X Y f U hf x
+    rw [Scheme.stalkMap_comp, hP.cancel_right_isIso]
+    exact hf _
+  · intro X Y f ι U hU hf x
+    have hy : x ∈ iSup U := by rw [hU]; trivial
+    obtain ⟨i, hi⟩ := Opens.mem_iSup.mp hy
+    rw [← hP.cancel_right_isIso _ ((U i).ι.stalkMap ⟨x, hi⟩)]
+    simpa [Scheme.stalkMap_comp] using hf i ⟨x, hi⟩
+
+lemma stalkwise_Spec_iff (hP : RingHom.RespectsIso P) {R S : CommRingCat} (φ : R ⟶ S) :
```

Comment:
Maybe `stalkwise_Spec_map_iff`? For some reason GH does not let me suggest changes.